### ISSUE TITLE
Can sqlmap be detected?

### ISSUE THREAD

**Issue by Tensha** [sqlmapproject/sqlmap#4110]
I have one story about a firewall (not sure if its the firewall but probably), who istant kill my session (asp.net), when im try to using sqlmap. All vulnerable url are "tryable" by logged user. But when i start sqlmap with --cookies=    or with -r file from burpsuite(post or get), my session was istantly killed(white page at refresh). I think is not tamper related issue, its another issue. Any idea how to avoiding this or this is the best firewall in the world? :P cant find nothing similar on the web.

**Comment by bahbah** [comment]
Could be the user agent - try setting a different one with  --user-agent

**Comment by Tensha** [comment]
I look some strange things and difference by using firefox or chrome(normally webpage navigation). Firewall block me in some page when i use firefox(latest on kali linux).

Tryed on sqlmap --mobile, but nothing change. --random-agent, nothing change. I will do some specific try user agent try and i will report, but i think isnt a solution.

**Comment by radinmoradi** [comment]
Ls

**Comment by Tensha** [comment]
> Ls

can you explain? Thanks

**Comment by radinmoradi** [comment]
#4110

**Comment by Tensha** [comment]
Update.

with --user-agent=opera or --user-agent=chrome

heuristic says is vulnerable(before this,it usually says me url was unstable), possible database mysql etc. But it doesnt found point of injection in the end. **Probably because firewall kill my session too** (checked just after i noticed heuristic test was positive).

Mh still noway

**Comment by stamparm** [comment]
If you have a firewall in between:

A) First problematic thing is the usage of `sqlmap` default User-agent, blocked by many firewalls
B) Second one is the "heuristic payload" used in WAF/IPS detection (https://github.com/sqlmapproject/sqlmap/blob/master/lib/core/settings.py#L604), which has been made to trigger majority of protection mechanisms in existence
C) sqlmap itself uses (too) problematic SQLi payloads, which have to work. Otherwise, you'll have big trouble in later stages. Hence, if you are able to "find" a SQLi manually, that doesn't mean that you'll be able to exploit it properly, particularly with sqlmap.

**Comment by hahahahahahahahahahahahahahahah** [comment]
it's a POST request? if yes, then try this thing: `--csrf-token`
